### DISCUSSION TITLE
Issue using persistent layout with Vue.extend component

### DISCUSSION THREAD

**Discussion by BrianLangevin** [inertiajs/inertia#167] [Help (Vue)]
When using TypeScript, Vue components must be defined using `Vue.extend` in order for types to be properly inferred. However, when `Vue.extend` is used on Inertia pages, the layout property is ignored and no layout is rendered.

Has anyone come up with a way to use TypeScript with Inertia pages and persistent layouts?

**Comment by TheDoctor0** [comment]
Yes, I even made [PR](https://github.com/inertiajs/inertia/pull/183) that solve this issue.

**Comment by applicazza** [comment]
@BrianLangevin @TheDoctor0 

This is how I've done it with original inertia-vue.

`Layouts/index.ts:`

```ts
import DefaultLayout from './DefaultLayout.vue';
import ProjectLayout from '@/inertia/Layouts/ProjectLayout.vue';

export function defaultLayout(h: any, page: any) {
    return h(DefaultLayout, [page]);
}

export function projectLayout(h: any, page: any) {
    return h(DefaultLayout, [h(ProjectLayout, [page])]);
}

```

`Components/Dashboard.vue:`

```jsx
<template>

    <div class="scrollable-container">

        <div class="scrollable"></div>

    </div>

</template>

<script lang="ts">
    import Component from 'vue-class-component';
    import Vue from 'vue';
    import {defaultLayout} from '../Layouts';

    @Component
    export default class Dashboard extends Vue {
        static layout = defaultLayout;
    }
</script>

<style scoped>

</style>

```

Probably you also need to add next shim:

```ts
declare module '*.vue' {
    import Vue from 'vue';
    // noinspection JSDuplicatedDeclaration
    export default Vue;
}
```

Hope this helps.

**Comment by gajosadrian** [comment]
Anybody kwows how to use `remember` yet? I have tried `static remember = 'from'` and it doesn't work 😕

### EDIT:
Ok, I fixed it by doing this:
**declarations.d.ts**

```typescript
declare module 'vue/types/options' {
  interface ComponentOptions<V extends Vue> {
    remember?: any;
  }
}
```

**Some Vue Component**
```typescript
@Component({
  remember: 'form',
})
export default class extends Vue {}
```
